fuel-relayer/src/pending_blocks.rs

```diff
@@ -0,0 +1,617 @@
+                "Bundle send pair {}:{} of blocks:",
+                parent.header.height, block.header.height
+            );
+            if let Err(error) = self.call_contract(&parent, &block, provider).await {
+                warn!("Commit fuel block failed: {}", error);
+                break;
+            }
+            parent = block;
+        }
+    }
+
+    /// Handle commited block from contract.
+    fn handle_block_commit_append(
+        &mut self,
+        block_root: Bytes32,
+        height: BlockHeight,
+        da_height: DaBlockHeight,
+    ) {
+        if self.pending_block_commits.is_empty() {
+            let lcffh = self.last_commited_finalized_fuel_height;
+            if lcffh + 1u64.into() != height {
+                error!("We have missing logs from LCFFH {lcffh} to new height {height}")
+            }
+            // no pending commits, add one
+            self.pending_block_commits
+                .push_front(PendingBlock::new_commited_block(
+                    da_height, height, block_root,
+                ));
+            return;
+        }
+
+        let front_height = self.pending_block_commits.front().unwrap().block_height;
+        let back_height = self.pending_block_commits.back().unwrap().block_height;
+
+        if height < back_height {
+            // This case means that we somehow skipped block commit and didnt receive it in expected order.
+            error!(
+                "Commited block {height} is lower then current lowest pending block {back_height}."
+            );
+        } else if height > front_height {
+            // check if we are lagging against da layer
+            if self.chain_height < height {
+                error!(
+                    "Our chain height: {} is lower then da layer height {height}",
+                    self.chain_height
+                );
+            }
+            // new block received. Happy path
+            if height == front_height + BlockHeight::from(1u64) {
+                self.pending_block_commits
+                    .push_front(PendingBlock::new_commited_block(
+                        da_height, height, block_root,
+                    ));
+            } else {
+                error!("Commited block height {height} should be only increased by one from current height {front_height}.");
+            }
+        } else {
+            // happens if reverted commit is again visible
+            // iterate over pending blocks and set reverted commit as commited.
+            for pending in self.pending_block_commits.iter_mut() {
+                if pending.block_height == height {
+                    if !pending.reverted {
+                        error!("We received block {height} commit that was not reverted")
+                    }
+                    pending.da_height = da_height;
+                    pending.reverted = false;
+                    pending.block_height = height;
+                    pending.block_root = block_root;
```

Comment:
unnecessary variable setting

```suggestion
                    pending.reverted = false;
```